### PR TITLE
[plugin-web-app] Add dynamic variables providing browser window dimen…

### PR DESCRIPTION
**Pull request by IBrun**
…sion

### CODE REVIEW

File: docs/modules/plugins/partials/dynamic-variables-ui.adoc [hunk 1]
```diff
@@ -57,3 +57,31 @@ ${source-code}
 ----
 Then `${source-code}` matches `.+Home.+`
 ----
+
+=== Browser window dimension
```
Comment (valfirst) [inline docs/modules/plugins/partials/dynamic-variables-ui.adoc:61]: it's not available for mobile native apps, but `dynamic-variables-ui.adoc` contains info about vars common for both web and native mobile apps

File: docs/modules/plugins/partials/dynamic-variables-ui.adoc [hunk 2]
```diff
@@ -57,3 +57,31 @@ ${source-code}
...
+
+The set of dynamic variables provides ability to access browser outer window height and outer window width.
```
Comment (valfirst) [inline docs/modules/plugins/partials/dynamic-variables-ui.adoc:63]: how can users understand what is `outer window`?

**Review by valfirst — CHANGES_REQUESTED** [review]

File: vividus-plugin-web-app/src/main/java/org/vividus/ui/web/variable/BrowserWindowWidthDynamicVariable.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.ui.web.variable;
+
+import org.vividus.selenium.IWebDriverProvider;
+import org.vividus.ui.variable.AbstractWebDriverDynamicVariable;
+
+public class BrowserWindowWidthDynamicVariable extends AbstractWebDriverDynamicVariable
```
Comment (ikalinin1) [inline vividus-plugin-web-app/src/main/java/org/vividus/ui/web/variable/BrowserWindowWidthDynamicVariable.java:22]: I propose to use a single variable, browser-window-size.

Usage examples:
${browser-window-size.width}
${browser-window-size.heigh}

cc @valfirst  @avinBar

File: docs/modules/plugins/partials/dynamic-variables-ui.adoc (hunk 2 above)
Comment (IBrun) [reply docs/modules/plugins/partials/dynamic-variables-ui.adoc:63]: You mean that we don't need to specify that there is outer window, don't you? Or there is wrong wording? Or we need to add additional information for user?

Comment (valfirst) [reply docs/modules/plugins/partials/dynamic-variables-ui.adoc:63]: I guess it would be enough just `window`

**Review by valfirst — APPROVED** [review]

File: vividus-plugin-web-app/src/main/java/org/vividus/ui/web/variable/AbstractBrowserWindowDimensionDynamicVariable.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+            ToIntFunction<Dimension> valueProvider)
+    {
+        this.webDriverManager = webDriverManager;
+        this.valueProvider = valueProvider;
+    }
+
+    @Override public DynamicVariableCalculationResult calculateValue()
+    {
+        return DynamicVariableCalculationResult.withValue(
+                String.valueOf(valueProvider.applyAsInt(webDriverManager.getSize())));
```
Comment (ikalinin1) [inline vividus-plugin-web-app/src/main/java/org/vividus/ui/web/variable/AbstractBrowserWindowDimensionDynamicVariable.java:41]: The current implementation doesn't take into account WebDriver initialisation

File: vividus-plugin-web-app/src/main/java/org/vividus/ui/web/variable/AbstractBrowserWindowDimensionDynamicVariable.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+            IGenericWebDriverManager webDriverManager, ToIntFunction<Dimension> valueProvider)
+    {
+        this.webDriverProvider = webDriverProvider;
+        this.webDriverManager = webDriverManager;
+        this.valueProvider = valueProvider;
+    }
+
+    @Override public DynamicVariableCalculationResult calculateValue()
+    {
+        if (webDriverProvider.isWebDriverInitialized())
```
Comment (valfirst) [inline vividus-plugin-web-app/src/main/java/org/vividus/ui/web/variable/AbstractBrowserWindowDimensionDynamicVariable.java:43]: what about reusing `AbstractWebDriverDynamicVariable` ?

**Review by valfirst — CHANGES_REQUESTED** [review]

File: vividus-plugin-web-app/src/main/java/org/vividus/ui/web/variable/AbstractBrowserWindowDimensionDynamicVariable.java (hunk 5 above)
Comment (IBrun) [reply vividus-plugin-web-app/src/main/java/org/vividus/ui/web/variable/AbstractBrowserWindowDimensionDynamicVariable.java:43]: Please give more details

Comment (valfirst) [reply vividus-plugin-web-app/src/main/java/org/vividus/ui/web/variable/AbstractBrowserWindowDimensionDynamicVariable.java]: `AbstractBrowserWindowDimensionDynamicVariable` extends `AbstractWebDriverDynamicVariable`

**Review by valfirst — APPROVED** [review]

**Review by uarlouski — APPROVED** [review]

**Pull request merged**